controller/producers.py

```diff
@@ -0,0 +1,17 @@
+
+from model.producer_abstract import ProducerAbstract
+
+
+class Producers(ProducerAbstract):
+
+    def __init__(self, producers):
+        self.producers = producers
+
+    def add(self, user_id):
+        """ function to add new producer"""
+        if user_id in self.producers:
+            return("user already added")
+        else:
+            self.user_id = user_id
```

Comment:
i don't understand, why do we have this line? producers should not have any property called user_id

Reply:
producers in this context is just a list containing the ids of users that have registered as producers. So, I am checking to see if a new producer is already registered before they are allowed to registered. 

True, producers does not have any property called user_id. writing "self.user_id" is redundant. I will just pass the "user_id" directly into the add method to add a new producer. 

I hope it makes better sense now.